addons/event/models/event_ticket.py

```diff
@@ -127,14 +128,31 @@ def _constrains_dates_coherency(self):
             if ticket.start_sale_datetime and ticket.end_sale_datetime and ticket.start_sale_datetime > ticket.end_sale_datetime:
                 raise UserError(_('The stop date cannot be earlier than the start date.'))
 
-    @api.constrains('seats_available', 'seats_max')
-    def _constrains_seats_available(self):
-        for record in self:
-            if record.seats_max and record.seats_available < 0:
-                raise ValidationError(
-                    _('No more available seats for the ticket %s (%s). '
-                      'Raise the limit or remove some other confirmed registrations first.',
-                      record.name, record.event_id.name))
+    @api.constrains('registration_ids', 'seats_max')
+    def _check_seats_availability(self):
+        for ticket in self:
+            if ticket.seats_max and ticket.seats_available < 0:
+                raise ValidationError(_(
+                    'Impossible to have a number of confirmed registrations above the seat limit '
+                    'for the ticket %(ticket_name)s (%(event_name)s). Raise the limit or remove '
+                    'some other confirmed registrations first.',
+                    ticket_name=ticket.name, event_name=ticket.event_id.name))
+
+    def name_get(self):
+        """Adds ticket seat availability if requested by context."""
+        if not self.env.context.get('name_with_seat_availability'):
+            return super().name_get()
+        res = []
+        for ticket in self:
+            if not ticket.seats_max:
+                availability = _('unlimited seats')
```

Comment:
Same remark about the context-based translation (aka if we update the other one, let us mimic it here :) )